js/msp/MSPHelper.js

```diff
@@ -957,7 +958,7 @@ MspHelper.prototype.process_data = function(dataHandler) {
             dataHandler.callbacks.splice(i, 1);
 
             // fire callback
-            if (callback) callback({'command': code, 'data': data, 'length': data.byteLength});
+            if (callback) callback({'command': code, 'data': data, 'length': data.byteLength, 'crcError': crcError});
```

Comment:
So to simplify…

Previous behavior: CRC error would result in the dataView being a null object and the callbacks would generate runtime errors because they tried to reference the object and had no idea that a crc error occurred.  This is the cause of the hang during blackbox downloads, but it would affect any of the msp message types.

Current patch:  Provides a crcError flag so the handlers can react appropriately and returns the received payload even though it has crc errors.  This works around the problem of the handlers referencing the null object, but this may not be a good idea as they may be acting on invalid data.  Previously they would crash, now they won't but would still have no idea that the data was invalid (except for blackbox downloading since it examines the crcError now).

Proposal: Revert the behavior of nulling the data view object and refactor the callback in `MspHelper.prototype.dataflashRead` to check the crcError flag before referencing the possibly null data object.  This would still allow any other message type to potentially crash on a crc error (the previous behavior) but eliminates the risk of them acting on potentially corrupt data.  Later each hander could be updated to examine the ecrError flag and act appropriately (drop the message or retry).